lxd/project/project.go

```diff
@@ -48,3 +53,38 @@ func StorageVolumeParts(projectStorageVolumeName string) (string, string) {
 	parts := strings.SplitN(projectStorageVolumeName, "_", 2)
 	return parts[0], parts[1]
 }
+
+// StorageVolumeProject returns the project name to use to for the volume based on the requested project.
+// For custom volume type, if the project specified has the "features.storage.volumes" flag enabled then the
+// project name is returned, otherwise the default project name is returned. For all other volume types the
+// supplied project name is returned.
+func StorageVolumeProject(c *db.Cluster, projectName string, volumeType int) (string, error) {
+	// Non-custom volumes always use the project specified.
+	if volumeType != db.StoragePoolVolumeTypeCustom {
+		return projectName, nil
+	}
+
+	var project *api.Project
+	var err error
+
+	err = c.Transaction(func(tx *db.ClusterTx) error {
+		project, err = tx.ProjectGet(projectName)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	})
+
+	if err != nil {
+		return "", errors.Wrapf(err, "Failed to load project %q", projectName)
+	}
+
+	// Custom volumes only use the project specified if the project has the features.storage.volumes feature
+	// enabled, otherwise the legacy behaviour of using the default project for custom volumes is used.
+	if project.Config["features.storage.volumes"] == "true" {
```

Comment:
That should be using `shared.IsTrue` as the user may have set it to `True` or `1` or something.

Reply:
Looks like that `== "true"` has been a bit of a pattern in our project code, we should fix that to line up with the rest of our config handling.

Reply:
@stgraber have fixed these existing `== "true"` with a couple of new commits.